### PR TITLE
[add] initial support of AI.TENSORGET command

### PR DESCRIPTION
**Pull request by filipecosta90**
- [add] Adds the Tensor class. 
- [add] initial support of AI.TENSORGET command
- [add] Adds JRedisAIRunTimeException class, which is thrown when RedisAI encounters a runtime error during command execution/parsing.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -89,8 +89,8 @@
 				<artifactId>maven-compiler-plugin</artifactId>
 				<version>3.1</version>
 				<configuration>
-					<source>1.7</source>
-					<target>1.7</target>
+					<source>8</source>
```
Comment (gkorland) [inline pom.xml:92]: @filipecosta90 why?

File: src/main/java/com/redislabs/redisai/RedisAI.java [hunk 2]
```diff
@@ -4,9 +4,10 @@
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.util.ArrayList;
+import java.util.*;
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/RedisAI.java:7]: @filipecosta90 why?

File: src/main/java/com/redislabs/redisai/RedisAI.java [hunk 3]
```diff
@@ -16,6 +17,8 @@
 import redis.clients.jedis.util.Pool;
 import redis.clients.jedis.util.SafeEncoder;
 
+import javax.xml.crypto.Data;
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/RedisAI.java:20]: @filipecosta90 is it needed?

File: src/main/java/com/redislabs/redisai/RedisAI.java [hunk 4]
```diff
@@ -94,7 +97,38 @@ public boolean setTensor(String key, Object tensor, int[] dimensions){
...
+  public Tensor getTensor(String key) throws JRedisAIRunTimeException {
+    try (Jedis conn = getConnection()) {
+      List<?> reply = sendCommand(conn, Command.TENSOR_GET, SafeEncoder.encode(key), Keyword.META.getRaw(), Keyword.VALUES.getRaw() ).getObjectMultiBulkReply();
+      if(reply.isEmpty()) {
+        return null;
+      }
+      if (reply.size() <6){
+        throw new JRedisAIRunTimeException("Expected a minimum Array Reply of size 6");
+      }
+      String dtypeString = SafeEncoder.encode((byte[]) reply.get(1));
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/RedisAI.java:116]: @filipecosta90 shouldn't you make this code more protocol safe and use the argument name in the reply for the parsing?

File: src/main/java/com/redislabs/redisai/Tensor.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    public long[] getShape() {
+        return shape;
+    }
+
+    public DataType getDataType() {
+        return dataType;
+    }
+
+    public void setDataType(DataType dataType) {
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/Tensor.java:30]: @filipecosta90 where are you using the setters? shouldn't we set all fields as final?

Comment (filipecosta90) [reply src/main/java/com/redislabs/redisai/Tensor.java:30]: @gkorland was planning to enable setting tensors via the tensorclass as well.

File: src/main/java/com/redislabs/redisai/RedisAI.java (hunk 2 above)
Comment (filipecosta90) [reply src/main/java/com/redislabs/redisai/RedisAI.java:7]: changed towards importing ArrayList and List specifically

File: src/main/java/com/redislabs/redisai/RedisAI.java (hunk 3 above)
Comment (filipecosta90) [reply src/main/java/com/redislabs/redisai/RedisAI.java:20]: no. probably imported wrongfully while testing approaches. changing

File: src/main/java/com/redislabs/redisai/RedisAI.java (hunk 4 above)
Comment (filipecosta90) [reply src/main/java/com/redislabs/redisai/RedisAI.java:116]: making it more future proof. will submit changes in next commit

File: pom.xml (hunk 1 above)
Comment (filipecosta90) [reply pom.xml:92]: moving to the previous config.

File: src/main/java/com/redislabs/redisai/RedisAI.java [hunk 6]
```diff
@@ -94,7 +96,55 @@ public boolean setTensor(String key, Object tensor, int[] dimensions){
...
+    try (Jedis conn = getConnection()) {
+      List<?> reply = sendCommand(conn, Command.TENSOR_GET, SafeEncoder.encode(key), Keyword.META.getRaw(), Keyword.VALUES.getRaw() ).getObjectMultiBulkReply();
+      if(reply.isEmpty()) {
+        return null;
+      }
+      DataType dtype = null;
+      long[] shape = null;
+      Object values = null;
+      Tensor tensor = null;
+      for (int i = 0; i < reply.size(); i+=2) {
```
Comment (filipecosta90) [inline src/main/java/com/redislabs/redisai/RedisAI.java:116]: @gkorland please revise this approach

Comment (gkorland) [reply src/main/java/com/redislabs/redisai/RedisAI.java:116]: why not switch on String? https://www.geeksforgeeks.org/string-in-switch-case-in-java/

Comment (filipecosta90) [reply src/main/java/com/redislabs/redisai/RedisAI.java:116]: :+1: changed in the new commit

**Pull request merged**